exporting/send_internal_metrics.c

```diff
@@ -3,16 +3,170 @@
 #include "exporting_engine.h"
 
 /**
- * Send internal metrics
+ * Create a chart for the main exporting thread CPU usage
+ *
+ * @param st_rusage the thead CPU usage chart
+ * @param rd_user a dimension for user CPU usage
+ * @param rd_system a dimension for system CPU usage
+ */
+void create_main_rusage_chart(RRDSET **st_rusage, RRDDIM **rd_user, RRDDIM **rd_system)
+{
+    if (*st_rusage && *rd_user && *rd_system)
+        return;
+
+    *st_rusage = rrdset_create_localhost(
+        "netdata", "exporting_main_thread_cpu", NULL, "exporting", NULL, "Netdata Main Exporting Thread CPU Usage",
+        "milliseconds/s", "exporting", NULL, 130600, localhost->rrd_update_every, RRDSET_TYPE_STACKED);
+
+    *rd_user = rrddim_add(*st_rusage, "user", NULL, 1, 1000, RRD_ALGORITHM_INCREMENTAL);
+    *rd_system = rrddim_add(*st_rusage, "system", NULL, 1, 1000, RRD_ALGORITHM_INCREMENTAL);
+}
+
+/**
+ * Send the main exporting thread CPU usage
+ *
+ * @param st_rusage a thead CPU usage chart
+ * @param rd_user a dimension for user CPU usage
+ * @param rd_system a dimension for system CPU usage
+ */
+void send_main_rusage(RRDSET *st_rusage, RRDDIM *rd_user, RRDDIM *rd_system)
+{
+    struct rusage thread;
+    getrusage(RUSAGE_THREAD, &thread);
```

Comment:
This function returns an integer, should we not test and report possible errors?

Reply:
I did it intentionally - it's the scope of the next planned issue #7173.

Reply:
Only two errors are possible in general, but not in this case (as well as in the other 31 places in the Netdata code). So, it doesn't make much sense to do an additional check.

Reply:
All right right, thank you!